src/android/IonicDeploy.java

```diff
@@ -147,8 +148,10 @@ private String generateUDID() {
 
   private Boolean isDebug() {
     try {
-      if ((this.myContext.getPackageManager().getPackageInfo(this.myContext.getPackageName(), 0).applicationInfo.flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0) {
-        return true;
+      if ((this.myContext.getPackageManager().getPackageInfo(this.myContext.getPackageName(), 0).applicationInfo.flags & ApplicationInfo.FLAG_DEBUGGABLE) != 0 ) {
+        if(this.shouldDebug != "false"){
```

Comment:
you could combine both if statements to one.

Reply:
Important: to compare a String. **Always** use .equals(). This code won't work.

Reply:
The idea here is that it doesnt have to be set. The only way it matters is if you set it to false.